language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -372,22 +372,40 @@ impl DiemVM {
             WriteSetPayload::Direct(change_set) => change_set.clone(),
             WriteSetPayload::Script { script, execute_as } => {
                 let mut tmp_session = self.0.new_session(remote_cache);
-                let args = convert_txn_args(script.args());
```

Comment:
Did this need to be inlined?

Reply:
I noticed that, too, but this changes it to match the existing code in `execute_script_or_script_function`. I somewhat prefer not inlining it, but it's good to be consistent.

I'm wondering if it would be better (perhaps as a separate PR that doesn't need to be cherry picked) to refactor this to share the common code with `execute_script_or_script_function`. It's fragile to copy-and-paste it like this.

Reply:
Yeah I'd definitely want to hold off on code sharing or what not to keep the PR as small as possible. The inlining of args was really just a copy-pasting effect. I think we should just match the code between the two call-sites as much as possible until we can do a bigger refactor of this